### PR TITLE
added port options support for preview and test commands

### PR DESCRIPTION
**Pull request by yjcxy12**
Connects #25 

- added `port` option inside `blackjack` command as options.
- added `port` option for `test` command. This is to solve the situation when using karma to run tests, if `preview` task is already running with port 9876, karma tests cannot run due to port being occupied.
- wrapped `server/bootstrap.js` in a function, and call that function with options in command `preview`, in order to pass down `port` option

### CODE REVIEW

File: src/commands/test.js [hunk 1]
```diff
@@ -13,6 +13,7 @@ const ENV_VARS = `BLACKJACK_HOME=${BLACKJACK_HOME} WORKING_DIR=${WORKING_DIR}`;
 
 export function call(options) {
   const runner = options.runner || 'mocha';
+  const port = options.port || 9875;
```
Comment (romidane) [inline src/commands/test.js]: Your repeat this port twice in your code. Surely you'd want this configuration back in the constants?

File: src/application.js [hunk 2]
```diff
@@ -26,11 +26,13 @@ export function run() {
     action(executeCommand);
 
   commander.command('preview').
+    option('-p, --port [port]', 'Which port do you wish to use to preview.').
```
Comment (joshnesbitt) [inline src/application.js]: Can we reword this from a question please?

File: src/application.js [hunk 3]
```diff
@@ -26,11 +26,13 @@ export function run() {
...
     description('Preview the built component locally using a development server.').
     action(executeCommand);
 
   commander.command('test').
     option('-r, --runner [type]', 'Which test runner you wish to run the tests with [mocha|karma].').
+    option('-p, --port [port]', 'Which port do you wish to use when running tests. (only used by karma)').
```
Comment (joshnesbitt) [inline src/application.js]: Can we reword this from a question please?

File: src/preview-server/bootstrap.js [hunk 4]
```diff
@@ -6,110 +6,112 @@ import logger from '../util/logger';
...
 } from '../constants';
 
-const COMPONENT_CONFIG = require(COMPONENT_CONF);
-const BLACKJACK_CONFIG = COMPONENT_CONFIG.blackjack || {};
-const PREVIEW_PATH = BLACKJACK_CONFIG.preview;
+export default function bootstrap(options) {
+  const COMPONENT_CONFIG = require(COMPONENT_CONF);
+  const BLACKJACK_CONFIG = COMPONENT_CONFIG.blackjack || {};
+  const PREVIEW_PATH = BLACKJACK_CONFIG.preview;
+  const PREVIEW_SERVER_PORT = options.port || 9876;
```
Comment (romidane) [inline src/preview-server/bootstrap.js]: You probably want this configuration back in a configuration file

File: src/application.js (hunk 3 above)
Comment (joshnesbitt) [inline src/application.js]: Maybe call this `karma-port` if it's only used by that?

**Comment by joshnesbitt** [comment]
Couple of comments but this is a nice change 👍

File: src/application.js [hunk 5]
```diff
@@ -26,11 +26,13 @@ export function run() {
     action(executeCommand);
 
   commander.command('preview').
+    option('-p, --port [port]', 'Which port you wish to use to preview.').
```
Comment (yjcxy12) [inline src/application.js]: Please feel free to come up with better wording?

File: src/application.js [hunk 6]
```diff
@@ -26,13 +26,13 @@ export function run() {
     action(executeCommand);
 
   commander.command('preview').
-    option('-p, --port [port]', 'Which port do you wish to use to preview.').
+    option('-p, --port [port]', 'Which port you wish to use to preview.').
```
Comment (joshnesbitt) [inline src/application.js]: Yup that's all I was referring to :)

**Comment by joshnesbitt** [comment]
@yjcxy12 is this ready for merge?

**Comment by yjcxy12** [comment]
I believe so.

**Comment by joshnesbitt** [comment]
Thanks @yjcxy12!

**Pull request merged**